Assets/HoloToolkit/Utilities/Scripts/BoundingBoxHelper.cs

```diff
@@ -0,0 +1,327 @@
+    private List<Vector3> worldBoundingCorners = new List<Vector3>();
+    private GameObject targetObject;
+    private bool rawBoundingCornersObtained = false;
+
+    /// <summary>
+    /// Objects that align to an target's bounding box can call this function in the object's UpdateLoop
+    /// to get current bound points;
+    /// </summary>
+    /// <param name="target"></param>
+    /// <param name="boundsPoints"></param>
+    /// <param name="ignoreLayers"></param>
+    public void UpdateNonAABoundingBoxCornerPositions(GameObject target, List<Vector3> boundsPoints, LayerMask ignoreLayers)
+    {
+        if (target != targetObject || rawBoundingCornersObtained == false)
+        {
+            GetRawBBCorners(target, ignoreLayers);
+        }
+
+        if (target == targetObject && rawBoundingCornersObtained)
+        {
+            boundsPoints.Clear();
+            for (int i = 0; i < rawBoundingCorners.Count; ++i)
+            {
+                boundsPoints.Add( target.transform.localToWorldMatrix.MultiplyPoint(rawBoundingCorners[i]) );
+            }
+
+            worldBoundingCorners.Clear();
+            worldBoundingCorners.AddRange(boundsPoints);
+        }
+    }
+
+    /// <summary>
+    /// This function gets the untransformed bounding box corner points of a GameObject.
+    /// </summary>
+    /// <param name="target"></param>
+    /// <param name="ignoreLayers"></param>
+    public void GetRawBBCorners(GameObject target, LayerMask ignoreLayers)
+    {
+        targetObject = target;
+
+        GameObject clone = GameObject.Instantiate(targetObject);
+        clone.transform.localRotation = Quaternion.identity;
+        clone.transform.position = Vector3.zero;
+        clone.transform.localScale = new Vector3(1, 1, 1);
+        Renderer[] renderers = clone.GetComponentsInChildren<Renderer>();
+        rawBoundingCorners.Clear();
+        rawBoundingCornersObtained = false;
+        for (int i = 0; i < renderers.Length; ++i)
+        {
+            var rendererObj = renderers[i];
+            if (ignoreLayers == (1 << rendererObj.gameObject.layer | ignoreLayers))
+            {
+                continue;
+            }
+            Vector3[] corners = null;
+            rendererObj.bounds.GetCornerPositionsFromRendererBounds(ref corners);
+            AddAABoundingBoxes(rawBoundingCorners, corners);
+        }
+
+        GameObject.Destroy(clone);
+
+        if (rawBoundingCorners != null && rawBoundingCorners.Count >= 4)
+        {
+            rawBoundingCornersObtained = true;
+        }
+    }
+
+    /// <summary>
+    /// this function gets the indices of the bounding cube corners that make up a face.
+    /// </summary>
+    /// <param name="index">the face index of the bounding cube 0-5</param>
+    /// <returns>an array of four integer indices</returns>
+    public int[] GetFaceIndices(int index)
+    {
+        switch (index)
+        {
+            case 0:
+                return new int[] { 0, 1, 3, 2 };
```

Comment:
we could make these readonly fields so we don't have to allocate new arrays each time.